### PR TITLE
Update error object in clrIfError context

### PR DESCRIPTION
**Pull request by gerinjacob**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Once the view is rendered, error object not getting updated in the context. This leads to error object to be out of sync.

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/src/icon/index.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 export * from './icon.element.js';
 export * from './icon.service.js';
+export { renderIcon } from './icon.renderer.js';
```
Comment (bbogdanov) [inline packages/core/src/icon/index.ts:9]: Why is that needed?

File: packages/core/src/icon/icon.stories.ts [hunk 2]
```diff
@@ -88,8 +88,8 @@ export const all = () => {
     'lg',
     propertiesGroup
   );
-  const dir = select(
-    'dir',
+  const direction = select(
```
Comment (bbogdanov) [inline packages/core/src/icon/icon.stories.ts]: Why is this change needed?

File: packages/core/src/icon/index.ts (hunk 1 above)
Comment (bbogdanov) [inline packages/core/src/icon/index.ts]: Why is this change needed?

**Review by bbogdanov — CHANGES_REQUESTED** [review]
Thank you for the contribution!

The commit message should be updated to `fix(angular:form): update error object in clrIfError context`

File: packages/core/src/icon/icon.stories.ts (hunk 2 above)
Comment (gerinjacob) [reply packages/core/src/icon/icon.stories.ts]: Sorry, undesired commits came along, fixed it.

File: packages/core/src/icon/index.ts (hunk 1 above)
Comment (gerinjacob) [reply packages/core/src/icon/index.ts]: Sorry, undesired commits came along, fixed it.

**Review by bbogdanov — APPROVED** [review]

**Comment by gnomeontherun** [comment]
Woops, I tried to change this to go against the `angular` branch and it automatically closed. Can you rebase your changes onto the `angular` branch?
